app/models/piece.rb

```diff
@@ -10,6 +10,57 @@ def validate_color?
     errors.add(:color, 'Invalid color') unless color == 'white' || color == 'black'
   end
 
+  def is_vertical?(x_destination, y_destination)
+  	self.x_position == x_destination
+  end
+
+  def is_horizontal?(x_destination, y_destination)
+  	self.y_position == y_destination
+  end
+
+  def is_diagonal?(x_destination, y_destination)
+  	x_destination - x_position == y_destination - y_position
+  end
+
+  def vertical_obstruction?(x_destination, y_destination)
+  	is_obstructed = false
+  	((self.y_position + 1)..y_destination).each do |y|
+	  	piece = Piece.where("game_id = ? AND x_position = ? AND y_position = ?", game_id, x_destination, y)
+      is_obstructed = piece.any?
+	  	break if is_obstructed
+		end
+		is_obstructed
+  end
+
+  def horizontal_obstruction?(x_destination, y_destination)
+  	is_obstructed = false
+  	((self.x_position + 1)..x_destination).each do |x|
+  		piece = Piece.where("game_id = ? AND x_position = ? AND y_position = ?", game_id, x, y_destination)
+  		is_obstructed = piece.any?
+  		break if is_obstructed
+  	end
+  	is_obstructed
+  end
+
+  def diagonal_obstruction?(x_destination, y_destination)
+  	is_obstructed = false
+  	y_pos = (self.y_position + 1)
+  	((self.x_position + 1)..x_destination).each do |x|
+  		piece = Piece.where("game_id = ? AND x_position = ? AND y_position = ?", game_id, x, y_pos)
```

Comment:
because a piece `belongs_to` ( [see code](https://github.com/queens-pawn/chess_app/blob/master/app/models/piece.rb#L2) ) a game, we can safely do
```
game.piece_at(x, y_pos)
```
which returns `nil` if there is no piece at given coordinates. making it possible to do

```
break if game.piece_at(x, y_pos)
```